### PR TITLE
docs: Add docs on linting

### LINKED ISSUE #15903: Tech debt: Clean up import declarations in Go code
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

To comply with `gofmt`, `go fmt`, `goimports`, and good practice,

**Imports declarations in the AWS Terraform Provider Go code should conform to these rules:**

1. Import declarations are grouped into 3 groups:
    - Short import path packages (i.e., built-in packages) (if any)
    - Long import path packages (if any)
    - Local import path packages (if any)
2. Declaration groups themselves should be in this order: short import path, long import path, local import paths.
3. Declaration groups should be separated by a single blank line.
4. Within each group, the lines should be in alphabetical order.

### Examples

**Bad Example 1.** This example is not grouped into short-path and long-path groups and the malformed groups are not alphabetized.

```go
import (
 	"fmt"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"regexp"
 	"testing"

 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 )
```

**Fixed Example 1.**

```go
import (
 	"fmt"
 	"regexp"
 	"testing"

 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 )
```

**Bad Example 2.** This example is not grouped into short-path and long-path groups, there are extra lines within the long-path group, and the malformed groups are not alphabetized.

```go
import (
 	"fmt"

 	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/aws/aws-sdk-go/aws"

 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"

 	"log"
 )
```

**Fixed Example 2.**

```go
import (
 	"fmt"
 	"log"

 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
```

### PR DESCRIPTION
**Pull request by YakDriver**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #15903

Adds documentation on linting Go code and correct import statements.

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

Output from acceptance testing: N/A

### CODE REVIEW

File: docs/contributing/pullrequest-submission-and-lifecycle.md [hunk 1]
```diff
@@ -70,7 +70,42 @@ based on these guidelines to speed up the review and merge process.
...
+
+The import statement in a Go file follows these rules (see [#15903](https://github.com/hashicorp/terraform-provider-aws/issues/15903)):
+
+1. Import declarations are grouped into a maximum of three groups:
+    - Standard packages (also called short import path or built-in packages)
+    - Third-party packages (also called long import path packages)
+    - Local packages
+2. Groups are in that order (i.e., standard packages, third-party packages, and local packages)
+3. Groups are separated by a single blank line
+4. Packages within each group are alphabetized
```
Comment (bflad) [inline docs/contributing/pullrequest-submission-and-lifecycle.md:100]: Use Markdown repeated `1.` syntax to remove number churn and combine 1 and 2:

```suggestion
1. Import declarations are grouped into a maximum of three groups with the following order:
    - Standard packages (also called short import path or built-in packages)
    - Third-party packages (also called long import path packages)
    - Local packages
1. Groups are separated by a single blank line
1. Packages within each group are alphabetized
```

File: docs/contributing/pullrequest-submission-and-lifecycle.md [hunk 2]
```diff
@@ -70,7 +70,42 @@ based on these guidelines to speed up the review and merge process.
...
+
+```console
+% make lint
+```
+
+`gofmt` will also fix many simple formatting issues for you:
+
+```console
+% gofmt -w -s ./aws
+```
```
Comment (bflad) [inline docs/contributing/pullrequest-submission-and-lifecycle.md:90]: We have a Makefile target to simplify this: `make fmt`

File: docs/contributing/pullrequest-submission-and-lifecycle.md [hunk 3]
```diff
@@ -70,7 +70,42 @@ based on these guidelines to speed up the review and merge process.
...
+    - Local packages
+2. Groups are in that order (i.e., standard packages, third-party packages, and local packages)
+3. Groups are separated by a single blank line
+4. Packages within each group are alphabetized
+
+Check your imports:
+
+```console
+% make importlint
+```
```
Comment (bflad) [inline docs/contributing/pullrequest-submission-and-lifecycle.md:106]: Not for this pull request, but we should also have a target to apply automatic fixes for imports.

Comment (YakDriver) [reply docs/contributing/pullrequest-submission-and-lifecycle.md:106]: I don't know of anything existing that fixes all the issues. `goimports` fixes some and `impi` identifies many but I ended up manually/regex fixing many issues.

**Review by bflad — APPROVED** [review]

**Pull request merged**